cmd/eval-dev-quality/cmd/evaluate.go

```diff
@@ -440,3 +444,69 @@ func (command *Evaluate) localRuntime(evaluationContext *evaluate.Context) (err
 
 	return nil
 }
+
+// dockerRuntime executes the evaluation for each model inside a docker container.
+func (command *Evaluate) dockerRuntime(ctx *evaluate.Context) (err error) {
+	// Filter all the args to pass them onto the container.
+	args := util.FilterArgs(os.Args[2:], []string{
+		"--runtime",
+		"--model",
+		"--result-path",
+	})
+
+	// Iterate over each model and start the container.
+	for _, model := range ctx.Models {
+		// We are skipping ollama models until we fully support pulling. https://github.com/symflower/eval-dev-quality/issues/100.
+		if ctx.ProviderForModel[model].ID() == "ollama" {
+			command.logger.Print("Skipping ollama model with docker runtime")
+
+			continue
+		}
+
+		// Create for each model a dedicated subfolder inside the results path.
+		resultPath, err := filepath.Abs(command.ResultPath)
+		if err != nil {
+			return err
+		}
+		if err := osutil.MkdirAll(resultPath); err != nil {
+			return err
+		}
```

Comment:
Isn't this creating the same folder over again? Shouldn't we join the model ID to the result path?

Reply:
Yeah that is outdated sry `MkdirAll` isn't necessary because we already created it. But what we need is `chmod` it to 777 - as we are running `evalaute` again inside the container with the result-path set there it is creating the folders accoridng to the subcommand.